sql-plugin/src/main/330+/scala/com/nvidia/spark/rapids/shims/GpuIntervalUtils.scala

```diff
@@ -557,4 +560,203 @@ object GpuIntervalUtils extends Arm {
       }
     }
   }
+
+  /**
+   * Cast day-time interval to string
+   * Rewrite from org.apache.spark.sql.catalyst.util.IntervalUtils.toDayTimeIntervalString
+   *
+   * @param micros     long micro seconds
+   * @param startField start field, valid values are [0, 3] indicates [DAY, HOUR, MINUTE, SECOND]
+   * @param endField   end field, should >= startField,
+   *                   valid values are [0, 3] indicates [DAY, HOUR, MINUTE, SECOND]
+   * @return ANSI day-time interval string, e.g.: interval '01 08:30:30.001' DAY TO SECOND
+   */
+  def toDayTimeIntervalString(
+      micros: ColumnVector,
+      startField: Byte,
+      endField: Byte): ColumnVector = {
+
+    val numRows = micros.getRowCount
+    val from = DT.fieldToString(startField).toUpperCase
+    val to = DT.fieldToString(endField).toUpperCase
+    val prefixStr = "INTERVAL '"
+    val postfixStr = s"' ${if (startField == endField) from else s"$from TO $to"}"
+
+    val retCv = withResource(new ArrayBuffer[ColumnVector]) { restHolder =>
+      // `restHolder` only hold one rest Cv;
```

Comment:
`rest` will be updated as following:

```
var rest = micros
days = rest / micros_per_day
old = rest
rest =  rest % micros_per_day.
old.close

......
seconds = rest / micros_per_second
old = rest
rest =  rest % micros_per_second
old.close
```

`var rest: Option` seems not work, because of Option can't reset the inner object. 
In order to close `rest cv` as soon as possible and put `rest cv` into a `withResource` block, 
I added a new `CloseableHolder` class to handle